.jenkins/Jenkinsfile

```diff
@@ -58,9 +58,16 @@ def AArch64GNUTest(String version, String build_type) {
             cleanWs()
             checkout scm
             def task = """
-                        cmake ${WORKSPACE} -G Ninja -DCMAKE_TOOLCHAIN_FILE=${WORKSPACE}/cmake/arm-cross.cmake -DCMAKE_BUILD_TYPE=${build_type} -DUSE_LIBSGX=OFF -Wdev
-                        ninja -v
-                        """
+                       curl ${DEVKITS_URI} | tar xvz -C ${WORKSPACE} --no-same-permissions
```

Comment:
This is liable to fail builds when the network has issues. Can we cache whatever exists at this URI?

Reply:
I'm admittedly not familiar with Jenkins. If I were to download this file outside of the container as part of the Jenkins step, will it remain on the system for subsequent runs? If so, I could download and extract the file only if it does not yet exist, then share it into the container.

Did you have something else in mind?

Reply:
Roping in @aserdean here...

Reply:
And @johnkord

Reply:
> I'm admittedly not familiar with Jenkins. If I were to download this file outside of the container as part of the Jenkins step that this line is embedded in, will it remain on the system for subsequent runs? If so, I could download and extract the file only if it does not yet exist, then share the extracted ones into the container. 

Tasks under `node` definitions in Jenkinsfiles (see `node("nonSGX")` in this example) are scheduled on a machine labelled `nonSGX` (there might be multiple machines with this label having free workers to execute the task).

There is no guarantee that subsequent runs will have the downloaded file. In fact, it's a good practice to start a CI stage from a clean workspace, and we try to do this. That's why we do `cleanWs()` above to clean the job workspace if it exists.

> Alternately, I could include a cached version inside the container image itself, and only download the file if a different one is specified via the corresponding pipeline argument.
> Did you have something else in mind?

I like this idea! I assume that most of the time the cached file will be used, and we will be fine. Whenever we want to update the cached file, we just update the `latest` CI Docker images.

Also, a simple retry logic on failure for the `curl` download command will work as well.

Reply:
See #1887.